### PR TITLE
Improved ACLE check

### PR DESCRIPTION
**Pull request by Un1q32**
Isolated a part of https://github.com/zlib-ng/zlib-ng/pull/1697 that I was looking forward too, since there's been no activity there for a while.  Fixes a false positive when building for armv6 Darwin with clang and possibly other systems as well, saves me from having to explicitly pass `--without-acle` to make the build work.

### CODE REVIEW

File: cmake/detect-intrinsics.cmake
Comment (Dead2) [inline cmake/detect-intrinsics.cmake:14]: **question:** 
Would it make sense to instead test for `-march=armv8-a+crc+simd` before falling back to `-march=armv8-a+crc` instead of preferring to not use the +simd option?
I am not intimately familiar with what the +simd option gives us in practice, but it seems logical to instead use the one without it as the fallback here.

Comment (Un1q32) [reply cmake/detect-intrinsics.cmake:14]: I'm not too familiar with cmake, the changes should be functionally the same as the ones to the configure script.

Comment (Dead2) [reply cmake/detect-intrinsics.cmake:14]: True enough. I don't know why it is like that there either. 🤷‍♂️

Comment (Un1q32) [reply cmake/detect-intrinsics.cmake:14]: Should I mark this as resolved or would you like me to switch the order of the flags?

Comment (Dead2) [reply cmake/detect-intrinsics.cmake:14]: I am not familiar enough with this, so I was hoping someone who is would add their thoughts on the matter.

The PR is also waiting for a review by someone else, I try not to merge things until someone else in addition to me has reviewed it, and especially when I am not feeling confident in whether it is correct or not.

Comment (nmoinvaz) [reply cmake/detect-intrinsics.cmake:14]: I'm not sure about it either. It does look like a simplification to some degree.

Comment (mtl1979) [reply cmake/detect-intrinsics.cmake:14]: Testing `crc+simd` first would mean we can't disable NEON support... Some compilers require both extensions to be enabled to be able to target either one because NEON uses different calling convention.

Comment (Dead2) [reply cmake/detect-intrinsics.cmake:14]: @mtl1979 Thanks, that makes sense. But do we take this into account when do the runtime tests as well?

And would it be possible for you to do a review of this PR? I'd like at least one other approval before I merge it. To me it looks ok, so I will merge it eventually even without any other reviews.

**Review by Dead2 — APPROVED** [review]

File: cmake/detect-intrinsics.cmake
Comment (mtl1979) [reply cmake/detect-intrinsics.cmake:14]: @Dead2 Runtime check isn't really affected as the issue is just with the command-line parameters... Some compilers don't enable FPU, which means NEON can't be used. By default FPU should be enabled when available.

I usually try to review every commit that changes something I have written or modified earlier, but sometimes I'm busy with other projects and sometimes I only have small remarks about inline comments or non-code text changes that don't look quite right. If there is something seriously wrong, I'm usually loud enough that everyone notice.

File: configure
Comment (mtl1979) [inline configure]: This is still ambiguous as all new intrinsics after ARMv7 are eventually implemented in both ARMv8 and subsequent processor families... "ARMv8 CRC intrinsics" would be less ambiguous.

File: cmake/detect-intrinsics.cmake
Comment (mtl1979) [inline cmake/detect-intrinsics.cmake]: "ARMv8 CRC intrinsics" would be less ambiguous, see my other comment...

**Pull request merged**
